### ISSUE TITLE
Freehand drawing doesn't work

### ISSUE THREAD

**Issue by mufar94** [mypaint/mypaint#537]
Tablet: Huion H610 Pro
Current MyPaint ver: 1.2.0 beta 3
Previous working ver: 1.2.0b - tumagonx

Details:
I tried opening both version at same time, seems I can't draw in freehand on ver 1.2.0 beta 3.

Tries:
I opened up 'Test input device' and checked 'visualise rendering', both version give response but for beta 3 ver. it suddenly work only on first stroke. On next stroke, it won't render any lines.

Other tools work properly so far. Mouse drawing works properly. 
Hopefully this can be fix so I'm able to use my tablet on future releases.

**Comment by achadwick** [comment]
Hi there. We have a few known issues with Huion tablets that I'm trying to get to the bottom of. Does the behaviour change at all if you close all of your open brush group tabs?

Possibly related: #514 (and just maybe, #527)

**Comment by mufar94** [comment]
I closed brush group tabs. It does not render anything. I also tried closing all tabs. It doesn't work.

There's something new I noticed:
1) I open up background chooser window. - The background chooser window is active.
2)The background chooser window is left open. I tried a line stroke, - background chooser window become inactive. A line rendered.
3) I tried another stroke, it does not render.
4) I make background chooser window active by clicking it. then I did as in #2. Same thing happens.

**Comment by mufar94** [comment]
There's appear to be two devices listed in Preference>Devices
http://imgur.com/6amjKoZ
It 'kind of' fix the problem when I disable one of them.
Although, when I restart Mypaint on next session, both listed was automatically set to 'ignore'.
So, I need to set one of them to 'any task' each time I start Mypaint.
Hope this might help.

**Comment by achadwick** [comment]
Inlining that for clarity. You can post images here on Github too, just drag them into the text box.

> ![imgur image](https://i.imgur.com/6amjKoZl.png)

Interesting. It's not wrong that there are duplicates, that's just the the way different ends of the pen or tablet "pad" or puck devices appear to GDK. But they usually have different names...

The fact that the names and the numbers of axes are the same may mean that at startup MyPaint gets confused. I want to see if there's more we can do to distinguish one of those devices from the other.

@mufar94 
Please try to run MyPaint from its console debug window and paste all the console lines into a Gist <https://gist.github.com/>. You'll need to share the link here. It's the lines that start with `INFO: gui.device: New device` that I'm interested in, for example:

```
DEBUG: gui.device: device-added 'USB Optical Mouse'
INFO: gui.device: New device 'USB Optical Mouse' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='0461', product='4d15')
DEBUG: gui.device: device-removed 'USB Optical Mouse'
```

You'll find the "MyPaint wXX (console debug)" launcher in the start menu if it's one of the developer releases from GitHub.

When you're testing, flip over the pen to use the eraser end, press buttons on your tablet pad, stroke the surface if it's a pen+touch, use the mouse. The Windows builds are sometimes reliant on discovering new devices when new events come in over the canvas due to a GDK limitation.

Note that the console debug launcher runs MyPaint very slowly :smile:

**Comment by mufar94** [comment]
https://gist.github.com/mufar94/04ad1f45b61fecb9b353

Some info that might be useful:
Pen has no eraser, tablet is not pen + touch.
At startup with no interaction, only mouse is listed.
If MyPaint window is active, when pen enters drawable area on tablet:
https://gist.github.com/mufar94/04ad1f45b61fecb9b353#file-gistfile1-txt-L179
Only one Huion device listed in 'Devices'. As soon as the pen touch the surface:
https://gist.github.com/mufar94/04ad1f45b61fecb9b353#file-gistfile1-txt-L208
Two Huion device listed.

In another startup, if MyPaint window is inactive,when pen enters drawable area on tablet no Huion device listed.As soon as the pen touch the surface, two Huion devices are listed.

Later, all buttons are pressed, it works as it should, but nothing is logged into console.

**Comment by achadwick** [comment]
Aha! These messages look familiar from the revert that addresses part of #500:

```
DEBUG: gui.freehand: Ignoring event from <gtk.gdk.DeviceWintab object at 0x10381
7d0 (GdkDeviceWintab at 0x28aa0c0)>
DEBUG: gui.freehand: Ignoring event from <gtk.gdk.DeviceWintab object at 0x10381
7d0 (GdkDeviceWintab at 0x28aa0c0)>
```
-- https://gist.github.com/mufar94/04ad1f45b61fecb9b353#file-gistfile1-txt-L210 onwards

The code that produces these was reverted in 7ce9ba0bff23a422a135ed447fdd6bf7f5204bf7, but that hasn't made it into any release yet.

**Comment by mufar94** [comment]
Fixed in Mypaint 1.2.0 beta 4. Thanks.

**Comment by achadwick** [comment]
@mufar94 Thanks for confirming!
